### PR TITLE
Feature/denomination

### PR DESCRIPTION
**Pull request by gokch**
## What is changed?
integrate denomination logic to types/denomination.go
improve readability of some config variables.

## Is there risk of fork? 
Unlikely happens. all config tested.

### CODE REVIEW

File: cmd/aergocli/util/unit_test.go [hunk 1]
```diff
@@ -4,50 +4,61 @@ import (
 	"math/big"
 	"testing"
 
+	"github.com/aergoio/aergo/types"
 	"github.com/stretchr/testify/assert"
 )
 
 func TestPasreUnit(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (gokch) [reply cmd/aergocli/util/unit_test.go]: changed 8471e91100b5abee0180b18d699952b9ade53c60

File: cmd/aergocli/util/unit_test.go [hunk 2]
```diff
@@ -4,50 +4,61 @@ import (
...
 	amount, err := ParseUnit("1 aergo")
 	assert.NoError(t, err, "parsing aergo")
-	t.Log(amount)
+	assert.Equalf(t, types.NewAmount(1, types.Aergo).String(), amount.String(), "parse is failed")
+
 	amount, err = ParseUnit("101 Aergo")
 	assert.NoError(t, err, "parsing Aergo")
-	t.Log(amount)
-	amount, err = ParseUnit("101 AERGO")
-	assert.NoError(t, err, "parsing AERGO")
```
[the remark this answers is not in the document]
Comment (gokch) [reply cmd/aergocli/util/unit_test.go:18]: fixed 8471e91100b5abee0180b18d699952b9ade53c60

File: cmd/aergocli/util/unit_test.go [hunk 3]
```diff
@@ -4,50 +4,61 @@ import (
...
+	assert.Equalf(t, types.NewAmount(1, types.Gaer).String(), amount.String(), "parse is failed")
+
 	amount, err = ParseUnit("1010")
-	assert.Equal(t, new(big.Int).SetUint64(1010), amount, "1010")
 	assert.NoError(t, err, "parsing implicit unit")
-	t.Log(amount)
+	assert.Equalf(t, types.NewAmount(1010, types.Aer).String(), amount.String(), "parse is failed")
 }
 
 func TestPasreDecimalUnit(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (gokch) [reply cmd/aergocli/util/unit_test.go]: fixed 8471e91100b5abee0180b18d699952b9ade53c60

File: contract/vm_dummy/vm_dummy_test.go [hunk 4]
```diff
@@ -574,8 +574,8 @@ func TestDeploy2(t *testing.T) {
 	defer bc.Release()
 
 	err = bc.ConnectBlock(
-		NewLuaTxAccount("user1", 100000000000000000),
-		NewLuaTxDeploy("user1", "deploy", 50000000000, code),
+		NewLuaTxAccount("user1", types.Aergo),
+		NewLuaTxDeploy("user1", "deploy", types.Aergo/2, code),
```
[the remark this answers is not in the document]
Comment (gokch) [reply contract/vm_dummy/vm_dummy_test.go]: fixed 8471e91100b5abee0180b18d699952b9ade53c60

**Review by kroggen — CHANGES_REQUESTED** [review]

**Review by kroggen — APPROVED** [review]

**Pull request merged**